src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -583,6 +583,7 @@ private void ProcessPingFrame(FrameHeader frameHeader)
             // Don't wait for completion, which could happen asynchronously.
             LogExceptions(SendPingAckAsync(_incomingBuffer.ActiveMemory.Slice(0, FrameHeader.PingLength)));
 
+            // SendPingAckAsync copies the buffer for us, so it is safe to not wait for completion.
```

Comment:
I think I read this code wrong when I initially reviewed it...

It seems better to just read the pingContent into a long here, in ProcessPingFrame, and then pass the resulting long into SendPingAckAsync. That's generally how these Process*Frame routines work, they read whatever data they need out of the frame. And it's more obviously correct.

Reply:
By reading the integer in `SendPingAckAsync`, it avoids depending on `ProcessPingFrame` (to have the same endianness scheme).

I agree that consistency with other Process*Frame is better here, so I've moved the read into `ProcessPingFrame` and added comments to note the dependency.